packages/playwright-core/src/client/page.ts

```diff
@@ -180,18 +185,23 @@ export class Page extends ChannelOwner<channels.PageChannel> implements api.Page
 
   private async _onRoute(route: Route, request: Request) {
     const routeHandlers = this._routes.slice();
-    for (const routeHandler of routeHandlers) {
-      if (!routeHandler.matches(request.url()))
-        continue;
-      if (routeHandler.willExpire())
-        this._routes.splice(this._routes.indexOf(routeHandler), 1);
-      const handled = await routeHandler.handle(route, request);
-      if (!this._routes.length)
-        this._wrapApiCall(() => this._disableInterception(), true).catch(() => {});
-      if (handled)
-        return;
+    this._routesInFlight.add(route);
+    try {
+      for (const routeHandler of routeHandlers) {
+        if (!routeHandler.matches(request.url()))
+          continue;
+        if (routeHandler.willExpire())
+          this._routes.splice(this._routes.indexOf(routeHandler), 1);
+        const handled = await routeHandler.handle(route, request);
+        if (!this._routes.length)
+          this._wrapApiCall(() => this._disableInterception(), true).catch(() => {});
+        if (handled)
+          return;
+      }
+      await this._browserContext._onRoute(route, request);
+    } finally {
+      this._routesInFlight.delete(route);
```

Comment:
`BrowserContext` also has some routes that are not handled directly from this method.